distributed/tests/test_scheduler.py

```diff
@@ -3052,24 +3051,12 @@ async def test_memory_no_workers(s):
     assert s.memory.managed == 0
 
 
-@gen_cluster(client=True, nthreads=[])
-async def test_memory_is_none(c, s):
-    """If Worker.heartbeat() runs before Worker.monitor.update(), then
-    Worker.metrics["memory"] will be None and will need special handling in
-    Worker.memory and Scheduler.heartbeat_worker().
+@gen_cluster(config={"distributed.admin.system-monitor.interval": "999s"})
+async def test_infrequent_sysmon(s, a, b):
+    """It doesn't matter how infrequently SystemMonitor.update() is called; there's
+    always one invocation before the first heartbeat.
     """
-    with mock.patch("distributed.system_monitor.SystemMonitor.update"):
-        async with Worker(s.address, nthreads=1) as w:
-            await c.wait_for_workers(1)
-            f = await c.scatter(123)
-            await w.heartbeat()
-            assert s.memory.process == 0  # Forced from None
-            assert s.memory.managed == 0  # Capped by process even if we do have keys
-            assert s.memory.managed_in_memory == 0
-            assert s.memory.managed_spilled == 0
-            assert s.memory.unmanaged == 0
-            assert s.memory.unmanaged_old == 0
-            assert s.memory.unmanaged_recent == 0
```

Comment:
When I wrote this test, there was no call to `SystemMonitor.update()` at the end of `SystemMonitor.__init__()`.